agent/command/generate.go

```diff
@@ -108,7 +108,7 @@ func (c *generateTask) Execute(ctx context.Context, comm client.Communicator, lo
 		func() (bool, error) {
 			generateStatus, err := comm.GenerateTasksPoll(ctx, td)
 			if err != nil {
-				return true, err
+				return false, err
```

Comment:
This won't let me comment below but it looks like we should remove the ShouldExit special casing since we're always now skipping the retry, and should remove from the struct:
https://github.com/evergreen-ci/evergreen/blob/bdc2f5e9a42213d2202aa8f4bbf38e4a9b1fe6cf/rest/route/task_generate.go#L99
(I think for backwards compatibility we probably want to do that in a different PR though)

Reply:
Sounds good! Made a new-eng ticket: https://jira.mongodb.org/browse/EVG-17561